eo-runtime/src/test/java/EOorg/EOeolang/EOcageTest.java

```diff
@@ -277,6 +272,147 @@ private static void writeTo(final Phi cage, final Phi obj) {
+        }
+
+        @Test
+        void throwsExceptionIfRecursion() {
+            final Phi cage = new EOcage(Phi.Φ);
+            writeTo(cage, cage);
+            Assertions.assertThrows(
+                ExAbstract.class,
+                new Dataized(cage)::take,
+                "We expect the exception to be thrown since we have recursion here"
+            );
+        }
+
+        @Test
+        void doesNotThrowExceptionIfSmallDepth() {
+            final EOcage cage = new EOcage(Phi.Φ);
+            writeTo(
+                cage,
+                new RecursiveDummi(MAX_DEPTH / 2, cage)
+            );
+            Assertions.assertDoesNotThrow(
+                () -> new Dataized(cage).take(),
+                String.format(
+                    "We expect that dataizing of nested cage which recursion depth is less than property %s = %s does not throw %s",
+                    PhTracedEnclosure.MAX_CAGE_RECURSION_PROPERTY_NAME,
+                    System.getProperty(PhTracedEnclosure.MAX_CAGE_RECURSION_PROPERTY_NAME),
+                    ExAbstract.class
+                )
+            );
+        }
+
+        /**
+         * The boundary case when the depth is equal to the maximum allowed.
+         */
+        @Test
+        void doesNotThrowExceptionIfMaxDepth() {
+            final EOcage cage = new EOcage(Phi.Φ);
+            writeTo(
+                cage,
+                new RecursiveDummi(MAX_DEPTH, cage)
+            );
+            Assertions.assertDoesNotThrow(
+                () -> new Dataized(cage).take(),
+                String.format(
+                    "We expect that dataizing of nested cage which recursion depth is equal to property %s = %s does not throw %s",
+                    PhTracedEnclosure.MAX_CAGE_RECURSION_PROPERTY_NAME,
+                    System.getProperty(PhTracedEnclosure.MAX_CAGE_RECURSION_PROPERTY_NAME),
+                    ExAbstract.class
+                )
+            );
+        }
+
+        @Test
+        void throwsExceptionIfBigDepth() {
+            final EOcage cage = new EOcage(Phi.Φ);
+            writeTo(
+                cage,
+                new RecursiveDummi(MAX_DEPTH  + 1, cage)
+            );
+            Assertions.assertThrows(
+                ExAbstract.class,
+                () -> new Dataized(cage).take(),
+                String.format(
+                    "We expect that dataizing of nested cage which recursion depth is more than property %s = %s does not throw %s",
+                    PhTracedEnclosure.MAX_CAGE_RECURSION_PROPERTY_NAME,
+                    System.getProperty(PhTracedEnclosure.MAX_CAGE_RECURSION_PROPERTY_NAME),
+                    ExAbstract.class
+                )
+            );
+        }
+
+        /**
+         * Recursive {@link Phi}.
+         * @since 0.1
+         */
+        private final class RecursiveDummi extends PhDefault implements Atom {
```

Comment:
@levBagryansky I believe it should be `Dummy`

Reply:
done

Reply:
@levBagryansky I meant that there's a typo in `Dummi`, it should be Y at the end